tests/app/data/test_schema_validation.py

```diff
@@ -41,15 +44,73 @@ def test_schema(self):
+            return ["Schema Validation Error! File [{}] does not validate against schema. Error [{}]".format(file, e)]
         except JSONDecodeError as e:
-            return "JSON Parse Error! Could not parse [{}]. Error [{}]".format(file, e)
+            return ["JSON Parse Error! Could not parse [{}]. Error [{}]".format(file, e)]
+
+
+    @staticmethod
+    def validate_schema_contains_valid_routing_rules(file, json_to_validate):
+
+        errors = []
+
+        for block in SchemaHelper.get_blocks(json_to_validate):
+            if 'routing_rules' in block and len(block['routing_rules']) > 0:
+                for rule in block['routing_rules']:
+                    is_goto_rule = 'goto' in rule and 'id' in rule['goto'].keys()
+                    if is_goto_rule:
+                        try:
+                            if rule['goto']['id'] == 'summary':
+                                continue
+
+                            SchemaHelper.get_block(json_to_validate, rule['goto']['id'])
+                        except StopIteration:
+                            errors.append("Schema Error. File[{}] Routing rule routes to invalid block [{}]".format(file, rule['goto']['id']))
+
+        return errors
+
+
+    @staticmethod
+    def validate_schema_containing_routing_rules_has_default_if_nots_all_answers_routed(file, json_to_validate):
+
+        errors = []
+
+        for block in SchemaHelper.get_blocks(json_to_validate):
+
+            if 'sections' in block:
+                for section in block['sections']:
+                    if 'questions' in section:
+                        for question in section['questions']:
+                            if 'answers' in question:
+                                for answer in question['answers']:
+                                    if answer['type'] == 'Radio' or answer['type'] == 'Checkbox':
+
+                                        if 'routing_rules' in block and len(block['routing_rules']) > 0:
+
+                                            options = []
+
+                                            for option in answer['options']:
+                                                options.append(option['value'])
+
+                                            for rule in block['routing_rules']:
+                                                if 'goto' in rule:
+
+                                                    if 'when' in rule['goto'].keys():
+                                                        when = rule['goto']['when']
+                                                        if 'id' in when and when['id'] == answer['id'] and when['value'] in options:
+                                                            options.remove(when['value'])
+                                                    else:
+                                                        options = []
+
+                                            if len(options) > 0 and len(options) != len(answer['options']):
```

Comment:
Might be nice to refactor to boolean variable no_routing_or_all_routes for clarity